cli/src/command.rs

```diff
@@ -475,6 +475,66 @@ pub fn run() -> Result<()> {
 			#[cfg(not(feature = "polkadot-native"))]
 			panic!("No runtime feature (polkadot, kusama, westend, rococo) is enabled")
 		},
+		Some(Subcommand::BenchmarkBlock(cmd)) => {
+			let runner = cli.create_runner(cmd)?;
+			let chain_spec = &runner.config().chain_spec;
+			set_default_ss58_version(chain_spec);
+
+			#[cfg(feature = "rococo-native")]
+			if chain_spec.is_rococo() || chain_spec.is_wococo() || chain_spec.is_versi() {
+				return Ok(runner.async_run(|mut config| {
+					let (client, _, _, task_manager) = service::new_chain_ops(&mut config, None)?;
+
+					if let polkadot_client::Client::Rococo(pd) = &*client {
+						Ok((cmd.run(pd.clone()).map_err(Error::SubstrateCli), task_manager))
+					} else {
+						unreachable!("Checked above; qed")
+					}
+				})?)
+			}
```

Comment:
```suggestion
			struct RunBlockBenchmark(frame_benchmarking_cli::BlockCmd);
			
			impl polkadot_client::ExecuteWithClient for RunBlockBenchmark {
	type Output = Result<(), Error>;

	fn execute_with_client<Client, Api, Backend>(self, client: Arc<Client>) -> Self::Output {
		 cmd.run(client).map_err(Error::SubstrateCli)   
		}
			}
			
							return Ok(runner.async_run(|mut config| {
					let (client, _, _, task_manager) = service::new_chain_ops(&mut config, None)?;

						Ok(RunBlockBenchmark(cmd).execute_with_client(client), task_manager)
				})?)
```

Reply:
Something like this.

Reply:
Good point. That would make it a lot less ugly.

Reply:
I tried to but something with the traits does not work. Probably missing `where` clause.  
I do a proper solution in https://github.com/paritytech/substrate/issues/11140.